applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -484,7 +484,11 @@ static bool event_handler(const struct event_header *eh)
 				discovering_peer_conn = NULL;
 			}
 			scan_stop();
-			scan_start();
+			/* ble_state keeps reference to connection object.
+			 * Cannot create new connection now.
+			 */
+			k_delayed_work_submit(&scan_start_trigger, 0);
+			scan_counter = SCAN_TRIG_TIMEOUT_MS;
```

Comment:
I am not 100% sure if we should change the counter. If user moves the keyboard scanning will disturb the cursor. Anyway there are pros and cons of this so let it be as is.